### PR TITLE
Add BME680 blocks, local compile refactor

### PR DESCRIPTION
**Pull request by MatzElectronics**
Also fixes a couple of small issues with custom code blocks

### CODE REVIEW

File: blockly/generators/propc.js [hunk 1]
```diff
@@ -337,7 +340,8 @@ Blockly.propc.finish = function (code) {
             definitions[def] = definitions[def].replace(/\{\{\$var_type_.*?\}\}/ig, "int").replace(/\{\{\$var_length_.*?\}\}/ig, '');
         }
 
-        if (def.indexOf(cCode) === -1) {  // exclude custom code blocks from these modifiers
+            console.log(def_names[def]);
```
Comment (zfi) [inline blockly/generators/propc.js]: This appears to be debugging code. Does it have another purpose?

File: blocklyc.js [hunk 2]
```diff
@@ -475,6 +476,7 @@ function cloudCompile(text, action, successHandler) {
                 } else {
                     var loadWaitMsg = (action !== 'compile') ? '\nDownload...' : '';
                     $("#compile-console").val($("#compile-console").val() + data['message'] + loadWaitMsg);
+    console.log(data);
```
Comment (zfi) [inline blocklyc.js]: This appears to be debugging code. Can this be safely removed?

File: blockly/generators/propc.js (hunk 1 above)
Comment (MatzElectronics) [reply blockly/generators/propc.js]: No, it can be removed.

File: blocklyc.js (hunk 2 above)
Comment (MatzElectronics) [reply blocklyc.js]: Same, this can be removed.

**Pull request merged**